field/FieldPointerUtilities.F90

```diff
@@ -1081,4 +1089,30 @@ subroutine assign_fptr_i8_rank1(x, fptr, rc)
       _RETURN(_SUCCESS)
    end subroutine assign_fptr_i8_rank1
 
+   subroutine check_typekind(field, typekind, rc)
+      type(ESMF_Field), intent(in) :: field
+      type(ESMF_TypeKind_Flag), intent(in) :: typekind
+      integer, optional, intent(out) :: rc
+      integer :: status
+      logical :: matches
+
+      matches = typekind_matches(field, typekind, _RC)
+      _ASSERT(matches, 'Field typekind does not match pointer type and kind.')
+      _RETURN(_SUCCESS)
+
+   end subroutine check_typekind
```

Comment:
Why 2 procedures?   I can understand that you don't want an assert here, so just put the assert on the uses above.  (Will add comments there now that I see this issue.)

Reply:
Ah.  Because the FieldGet call might fail for unrelated reasons.  Sigh.

Reply:
Still seems overkill to have 2 procedures here.  What is the downside of putting the assert in the other routine and deleting this one?